scripts/provision_health_checks.py

```diff
@@ -14,71 +14,76 @@
 logger = logging.getLogger(__name__)
 
 
-def main(argv):
-    parser = argparse.ArgumentParser(description='Dynamically reference or dereference Azul\'s Route53 health checks '
-                                                 'to/from composite health checkers not managed by Azul.')
-    group = parser.add_mutually_exclusive_group()
-    group.add_argument('--reference', '-R', dest='reference', action='store_true', default=False,
-                       help='Link Azul managed Route53 health check resources to DCP wide composite health check.')
-    group.add_argument('--dereference', '-D', dest='reference', action='store_false', default=False,
-                       help='Unlink Azul managed Route53 health check resources to DCP wide composite health check.')
-    options = parser.parse_args(argv)
-    health_checks = get_health_checks()
-    try:
-        dcp_wide_health_check = health_checks[f'dcp-health-check-{config.deployment_stage}']
-    except KeyError:
-        logger.info(f'DCP wide health check does not exist for {config.deployment_stage}')
-    else:
-        provision_health_check(health_checks, dcp_wide_health_check, options.reference)
+class ProvisionHealthChecks:
 
+    def __init__(self):
+        self.client = boto3.client('route53')
 
-def get_health_checks():
-    client = boto3.client('route53')
-    paginator = client.get_paginator('list_health_checks')
-    health_checks = chain.from_iterable(page['HealthChecks'] for page in paginator.paginate())
-    health_check_batches = more_itertools.chunked(health_checks, 10)
-    dcp_health_checks = {}
-    for health_check_batch in health_check_batches:
-        health_check_batch = {health_check['Id']: health_check for health_check in health_check_batch}
-        response = client.list_tags_for_resources(ResourceType='healthcheck',
-                                                  ResourceIds=list(health_check_batch.keys()))
-        for tag_sets in response['ResourceTagSets']:
-            assert tag_sets['ResourceType'] == 'healthcheck'
-            for tag in tag_sets['Tags']:
-                if tag['Key'] == 'Name':
-                    health_check_name = tag['Value']
-                    health_check_id = tag_sets['ResourceId']
-                    dcp_health_checks[health_check_name] = health_check_batch[health_check_id]
-    return dcp_health_checks
-
-
-def provision_health_check(health_checks, dcp_health_check, reference):
-    dcp_health_check_children = dcp_health_check['HealthCheckConfig']['ChildHealthChecks']
-    azul_health_checks = []
-    for resource_name in (config.service_name, config.indexer_name, config.data_browser_name, config.data_portal_name):
+    def main(self, argv):
```

Comment:
I would suggest leaving `main()` as a function instead of a member of the class. That leaves the class as a self contained Python interface to the provisioning, and `main` exists separately to add a CLI.

This pattern is used in https://github.com/DataBiosphere/azul/blob/develop/scripts/provision_credentials.py and https://github.com/DataBiosphere/azul/blob/develop/scripts/upload-tsv-files.py.

In those two particular cases, `main` is not a separate function, but I think it's more than practical to keep it separate.